### PR TITLE
[Bug] Check login status before calling autoSaveForm

### PR DESCRIPTION
**Pull request by Mottie**
## Description

I recently added form submission state to the save-in-progress (SiP) data, but this required the SiP data to be save upon submission. There are a few forms that don't require the user to be logged in to submit and thus throw a JS error since there is no user to associate the SiP data to.

This PR checks if the user is logged in, and if they are, it will then save the SiP data.

Related:
- Bug: https://dsva.slack.com/archives/CBU0KDSB1/p1605018423415400
- PR: https://github.com/department-of-veterans-affairs/vets-website/pull/14807
- Ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/15642

## Testing done

Added unit tests for new log in checks

## Screenshots

image
image (1)

## Acceptance criteria
- [ ] SiP data is not auto-saved when the user is not logged in
- [ ] No JS error

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by Mottie** [comment]
Included a `yarn.lock` update

File: src/platform/forms-system/src/js/review/SubmitController.jsx [hunk 1]
```diff
@@ -81,15 +82,25 @@ class SubmitController extends Component {
       this.props.setSubmission('status', 'validationError');
       this.props.setSubmission('hasAttemptedSubmit', true);
 
-      // Update save-in-progress with failed submit
-      submissionData.errors = errors;
-      this.props.autoSaveForm(formId, data, version, returnUrl, submissionData);
+      if (isLoggedIn) {
```
Comment (cvalarida) [inline src/platform/forms-system/src/js/review/SubmitController.jsx:85]:
```suggestion
      if (isLoggedIn && formConfig.prefillEnabled) {
```

**Review by cvalarida — APPROVED** [review]
I know we don't use `prefillEnabled` right now, but we really should, now that more form apps need to disable it. With that addition, I think it's solid. :+1: 

Bonus points for adding another test to check `prefillEnabled`, though. :wink:

File: src/platform/forms-system/src/js/review/SubmitController.jsx (hunk 1 above)
Comment (Mottie) [reply src/platform/forms-system/src/js/review/SubmitController.jsx]: Thanks! Added! :D

File: src/platform/forms-system/src/js/review/SubmitController.jsx [hunk 2]
```diff
@@ -45,8 +45,9 @@ class SubmitController extends Component {
   };
 
   handleSubmit = () => {
-    const { form, formConfig, pageList, trackingPrefix } = this.props;
+    const { form, formConfig, pageList, trackingPrefix, user } = this.props;
     const { formId, data, submission } = form;
+    const isLoggedIn = user?.login?.currentlyLoggedIn;
```
Comment (U-DON) [inline src/platform/forms-system/src/js/review/SubmitController.jsx:50]: There's a selector for this in `src/platform/user/selectors.js` that can be used in the `mapStateToProps`.

**Review by U-DON — APPROVED** [review]

**Pull request merged**